osquery/events/eventsubscriberplugin.h

```diff
@@ -0,0 +1,293 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#pragma once
+
+#include <gtest/gtest_prod.h>
+
+#include <osquery/core/plugins/plugin.h>
+#include <osquery/core/tables.h>
+#include <osquery/database/database.h>
+#include <osquery/events/eventer.h>
+#include <osquery/events/types.h>
+#include <osquery/utils/mutex.h>
+
+namespace osquery {
+
+class EventSubscriberPlugin : public Plugin, public Eventer {
+ public:
+  /**
+   * @brief Add Subscription%s to the EventPublisher this module will act on.
+   *
+   * When the EventSubscriber%'s `init` method is called you are assured the
+   * EventPublisher has `setUp` and is ready to subscription for events.
+   */
+  virtual Status init();
+
+  /// This is a plugin type and must implement a call method.
+  virtual Status call(const PluginRequest&, PluginResponse&) override;
+
+ protected:
+  /**
+   * @brief Store parsed event data from an EventCallback in a backing store.
+   *
+   * This method stores a single event
+   *
+   * @param r The row to add
+   *
+   * @return Was the element added to the backing store.
+   */
+
+  // clang-format off
+  [[deprecated("Group events together and use addBatch() instead.")]]
+  // clang-format on
+  Status
+  add(const Row& r);
```

Comment:
nitpick, remove the newline within the function signature

Reply:
I tried to remove this newline but clang-format/make format will restore it, probably due to how the [[deprecated]] tag is handled